iris-mpc-cpu/src/execution/hawk_main.rs

```diff
@@ -0,0 +1,174 @@
+use crate::{
+    database_generators::generate_galois_iris_shares,
+    execution::{
+        local::generate_local_identities,
+        player::{Role, RoleAssignment},
+        session::{BootSession, Session, SessionId},
+    },
+    hawkers::aby3_store::{Aby3Store, SharedIrisesRef},
+    hnsw::HnswSearcher,
+    network::grpc::{GrpcConfig, GrpcNetworking},
+    proto_generated::party_node::party_node_server::PartyNodeServer,
+    protocol::ops::setup_replicated_prf,
+};
+use aes_prng::AesRng;
+use clap::Parser;
+use eyre::Result;
+use hawk_pack::graph_store::GraphMem;
+use iris_mpc_common::iris_db::db::IrisDB;
+use itertools::{izip, Itertools};
+use rand::SeedableRng;
+use std::{sync::Arc, time::Duration};
+use tokio::task::JoinSet;
+use tonic::transport::Server;
+
+#[derive(Parser)]
+pub struct HawkArgs {
+    #[clap(short, long)]
+    party_index: usize,
+}
+
+pub async fn hawk_main(args: HawkArgs) -> Result<()> {
+    // ---- Shared setup ----
+
+    let n_parties = 3;
+    let port_start = 40000;
```

Comment:
I think if you add more async tests, then these ports will be congested. So I'd add addresses/ports to `HawkArgs` and use [this func](https://github.com/worldcoin/iris-mpc/blob/fb24b393893352996a9fd622609780b1403aea3b/iris-mpc-cpu/src/execution/local.rs#L25) to generate them for tests.

Reply:
Done!

Reply:
Done!